admin-dev/themes/new-theme/js/components/modal/confirm-modal.ts

```diff
@@ -0,0 +1,149 @@
+  customButtons: Array<HTMLButtonElement | HTMLAnchorElement>;
+}
+export type InputConfirmModalParams = Partial<ConfirmModalParams>;
+
+/**
+ * This class is used to build the modal DOM elements, it is not usable as is because it doesn't even have a show
+ * method and the elements are created but not added to the DOM. It just creates a basic DOM structure of a
+ * Bootstrap modal, thus keeping the logic class of the modal separated.
+ *
+ * This container is built on the basic ModalContainer and adds some confirm/cancel buttons along with a message
+ * in the body, it is mostly used as a Rich confirm dialog box.
+ */
+export class ConfirmModalContainer extends ModalContainer implements ConfirmModalContainerType {
+  footer!: HTMLElement;
+
+  closeButton!: HTMLElement;
+
+  confirmButton!: HTMLButtonElement;
+
+  /* This constructor is important to force the input type but ESLint is not happy about it*/
+  /* eslint-disable no-useless-constructor */
+  constructor(params: ConfirmModalParams) {
+    super(params);
+  }
+
+  buildModalContainer(params: ConfirmModalParams): void {
+    super.buildModalContainer(params);
+
+    // Modal message element
+    this.message.classList.add('confirm-message');
+    this.message.innerHTML = params.confirmMessage;
+
+    // Modal footer element
+    this.footer = document.createElement('div');
+    this.footer.classList.add('modal-footer');
+
+    // Modal close button element
+    this.closeButton = document.createElement('button');
+    this.closeButton.setAttribute('type', 'button');
+    this.closeButton.classList.add('btn', 'btn-outline-secondary', 'btn-lg');
+    this.closeButton.dataset.dismiss = 'modal';
+    this.closeButton.innerHTML = params.closeButtonLabel;
+
+    // Modal confirm button element
+    this.confirmButton = document.createElement('button');
+    this.confirmButton.setAttribute('type', 'button');
+    this.confirmButton.classList.add(
+      'btn',
+      params.confirmButtonClass,
+      'btn-lg',
+      'btn-confirm-submit',
+    );
+    this.confirmButton.dataset.dismiss = 'modal';
+    this.confirmButton.innerHTML = params.confirmButtonLabel;
+
+    // Appending element to the modal
+    this.footer.append(this.closeButton, ...params.customButtons, this.confirmButton);
+    this.content.append(this.footer);
+  }
+}
+
+/**
+ * ConfirmModal component
+ *
+ * @param {InputConfirmModalParams} params
+ * @param {Function} confirmCallback @deprecated You should rely on the confirmCallback param
+ * @param {Function} cancelCallback @deprecated You should rely on the closeCallback param
+ */
+export class ConfirmModal extends Modal implements ConfirmModalType {
+  modal!: ConfirmModalContainerType;
+
+  constructor(
+    inputParams: InputConfirmModalParams,
+    confirmCallback: (event: Event) => void,
+    cancelCallback = () => true,
+  ) {
+    const params: ConfirmModalParams = {
+      id: 'confirm-modal',
+      confirmMessage: 'Are you sure?',
```

Comment:
We'll need to translate these strings later.

Reply:
It is already handled! But not in the JS component, it's the provided configuration that must be translated and most of the type it is handled by a data attribute

Example:
https://github.com/PrestaShop/PrestaShop/blob/17b8b24937c98bfefa18e02eb2af7c1da70f70ea/admin-dev/themes/new-theme/js/pages/order/view/order-product-add.ts#L347
https://github.com/PrestaShop/PrestaShop/blob/2dc61fba9db8b531b071877f799a3f95c7978264/templates/bundles/PrestaShopBundle/Admin/Sell/Order/Order/Blocks/View/add_product_row.html.twig#L67

There is no need to handle internal translation inside this component as it would only complexify it